src/vaadin-form-layout.html

```diff
@@ -470,29 +463,34 @@
           }
 
           let col = 0;
-          Array.from(this.children).forEach((child) => {
-            if (getComputedStyle(child).display !== 'none') {
-              let colspan;
-
+          Array.from(this.children)
+            .filter(child => child.localName === 'br' || getComputedStyle(child).display !== 'none')
+            .forEach((child, index, children) => {
               if (child.localName === 'br') {
-                colspan = this._columnCount - col;
-              } else {
-                colspan = this._naturalNumberOrOne(parseFloat(child.getAttribute('colspan')));
+                // Reset column count on line break
+                col = 0;
+                return;
+              }
+
+              const nextIndex = index + 1;
+              const nextLineBreak = nextIndex < children.length && children[nextIndex].localName === 'br';
 
-                // Never span further than the number of columns
-                colspan = Math.min(colspan, this._columnCount);
+              if (nextLineBreak) {
+                const colspanRatio = (this._columnCount - col - 1) / this._columnCount;
+                child.style.setProperty(marginEndProp, `calc(${colspanRatio * containerWidth}px + ${colspanRatio} * ${columnSpacing})`);
```

Comment:
The formula here should take the colspan of the current child into account. Currently, in `colspanRatio` formula, the colspan of the current child is assumed to be 1.

Reply:
Updated. Test is added